taichi/program/sparse_matrix.h

```diff
@@ -57,7 +58,9 @@ class SparseMatrix {
   virtual ~SparseMatrix() = default;
 
   virtual void build_triplets(void *triplets_adr){};
-
+  virtual SparseMatrix &build_from_ndarray(const Ndarray &ndarray) {
+    return *this;
+  }
```

Comment:
This seems a bit weird to me.. Should `SparseMatrix` be an abstract class?

Reply:
Thanks,  `build_from_ndarray` is moved outside of `SparseMatrix` in the new commit.